workspaces/global-header/plugins/global-header/src/components/GlobalHeaderComponent.tsx

```diff
@@ -0,0 +1,148 @@
+
+/**
+ * @public
+ * Global Header Component properties
+ */
+export interface GlobalHeaderComponentProps {
+  globalHeaderMountPoints: GlobalHeaderComponentMountPoint[];
+  supportUrl: string;
+}
+
+export const GlobalHeaderComponent = ({
+  globalHeaderMountPoints,
+  supportUrl,
+}: GlobalHeaderComponentProps) => {
+  const { menuStates, handleOpen, handleClose } = useDropdownManager();
+  const { matchesFrontendRoute, shouldDisplaySupportIcon } =
+    useGlobalHeaderConfig();
+
+  const getDropdownButtonProps = (key: string) => ({
+    handleMenu: handleOpen(key),
+    anchorEl: menuStates[key],
+    setAnchorEl: handleClose(key),
+  });
+
+  const getIconButtonProps = (props: Record<string, any>) => {
+    const buttonPros = {
+      icon: props.icon ?? '',
+      tooltip: props.tooltip ?? '',
+      to: props.to ?? '',
+    };
+
+    if (props.icon === 'support' && buttonPros.to === '') {
+      buttonPros.to = supportUrl ?? '';
+    }
+    return buttonPros;
+  };
+  const filteredAndSortedGlobalHeaderComponents = useMemo(() => {
+    if (!globalHeaderMountPoints) {
+      return [];
+    }
+
+    const filteredAndSorted = globalHeaderMountPoints.filter(
+      component => (component.config?.priority ?? 0) > -1,
+    );
+
+    filteredAndSorted.sort(
+      (a, b) => (b.config?.priority ?? 0) - (a.config?.priority ?? 0),
+    );
+
+    return filteredAndSorted;
+  }, [globalHeaderMountPoints]);
+
+  const globalHeaderStartComponentsMountPoints =
+    filteredAndSortedGlobalHeaderComponents.filter(
+      component => component.config?.slot === Slot.HEADER_START,
+    );
+
+  const globalHeaderEndComponentsMountPoints =
+    filteredAndSortedGlobalHeaderComponents.filter(
+      component => component.config?.slot === Slot.HEADER_END,
+    );
+
+  const renderComponents = (mountPoints: GlobalHeaderComponentMountPoint[]) =>
+    mountPoints.map((mp, index) => {
+      const to = mp.config?.props?.to;
+      const icon = mp.config?.props?.icon;
+      const isExternal = to && isExternalUrl(to);
+      const isInternalRoute = to && matchesFrontendRoute(to);
+      const shouldShowSupportIcon = shouldDisplaySupportIcon(icon, to);
+
+      const displayHeaderIcon =
+        isExternal || isInternalRoute || shouldShowSupportIcon;
+
+      switch (mp.config?.type) {
+        case ComponentType.SEARCH:
+          return (
+            <ErrorBoundary>
+              <mp.Component key={`header-component-${index.toString()}`} />
+            </ErrorBoundary>
+          );
+        case ComponentType.DROPDOWN_BUTTON:
+          return (
+            <ErrorBoundary>
+              <mp.Component
+                key={`header-component-${index.toString()}`}
+                {...getDropdownButtonProps(mp.config?.key ?? index.toString())}
+                {...mp.config?.props}
+              />
+            </ErrorBoundary>
+          );
+        case ComponentType.ICON_BUTTON:
+          return (
+            <ErrorBoundary>
```

Comment:
I saw some react warnings:

> Warning: Each child in a list should have a unique "key" prop.
> Check the render method of `GlobalHeaderComponent`. See https://reactjs.org/link/warning-keys for more information.

I guess these warnings comes from here where the error boundaries needs a key.